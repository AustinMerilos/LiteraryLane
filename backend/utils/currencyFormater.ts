export default function currencyFormater(amount = 0) {
  // Define formatting options
  const options = {
    style: "currency",
    currency: "USD",
    minimumFractionDigits: 2,
    maximumFractionDigits: 2, // Ensure exactly 2 decimal places
  };

  // Create an Intl.NumberFormat instance with the specified options
  const formatter = Intl.NumberFormat("en-US", options);

  // Format the amount and return the result
  const formattedAmount = formatter.format(amount / 100);

  // Append ".00" if it's a clean dollar amount
  return formattedAmount.replace(/\.\d$/, ".00");
}
