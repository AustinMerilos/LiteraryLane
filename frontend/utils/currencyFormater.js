export default function currencyFormatter(amount = 0) {
  // Define formatting options
  const options = {
    style: "currency",
    currency: "USD",
    minimumFractionDigits: 2,
    maximumFractionDigits: 2, // Ensure exactly 2 decimal places
  };

  // Create an Intl.NumberFormat instance with the specified options
  const formatter = Intl.NumberFormat("en-US", options);

  // Format the amount and return the result without the currency symbol
  const formattedAmount = formatter.format(amount / 100);

  // Remove the currency symbol and return the formatted amount
  return formattedAmount.replace(/^\D+/, "");
}
