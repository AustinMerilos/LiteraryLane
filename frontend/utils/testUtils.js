import casual from "casual";
import { PAGINATION_QUERY } from "../components/pagination";

// seed it so we get consistent results
casual.seed(777);

const fakeItem = () => ({
  // __typename: 'Item',
  id: "abc123",
  price: 5000,
  user: null,
  photo: {
    id: "abc123",
    altText: "test item",
    image: {
      publicUrlTransformed: "test.jpg",
    },
  },
  name: "test item",
  description: "test",
});

const fakeUser = (overrides) => ({
  __typename: "User",
  id: "1234",
  name: casual.name,
  email: casual.email,
  permissions: ["ADMIN"],
  orders: [],
  cart: [],
  ...overrides,
});

const fakeOrderItem = () => ({
  __typename: "OrderItem",
  id: casual.uuid,
  image: {
    image: `${casual.word}.jpg`,
  },
  name: casual.words(),
  price: 1234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: "Order",
  id: "ord123",
  charge: "ch_123",
  total: 40000,
  items: [fakeOrderItem(), fakeOrderItem()],
  createdAt: "2024-01-11T20:16:13.797Z",
  user: fakeUser(),
});

const fakeCartItem = (overrides) => ({
  __typename: "CartItem",
  id: "abc123",
  quantity: 4,
  product: fakeItem(),
  user: fakeUser(),
  ...overrides,
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

function makePaginationMocksFor(length) {
  return [
    {
      request: { query: PAGINATION_QUERY },
      result: {
        data: {
          _allProductsMeta: {
            count: length,
          },
          itemsConnection: {
            __typename: "aggregate",
            aggregate: {
              count: length,
              __typename: "count",
            },
          },
        },
      },
    },
  ];
}

export {
  makePaginationMocksFor,
  LocalStorageMock,
  fakeItem,
  fakeUser,
  fakeCartItem,
  fakeOrder,
  fakeOrderItem,
};
