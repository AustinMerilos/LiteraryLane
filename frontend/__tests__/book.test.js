import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import { fakeItem } from "../utils/testUtils";
import Book from "../pages/book/book";
import React from "react";
import "@testing-library/jest-dom";

const product = fakeItem();

describe("<Book/>", () => {
  it("renders out the price tag and title", () => {
    const { container, debug } = render(
      <MockedProvider>
        <Book product={product} />
      </MockedProvider>
    );

    const priceTag = screen.getByText("50.00");
    expect(priceTag).toBeInTheDocument();
    const link = container.querySelector("a");
    expect(link).toHaveAttribute("href", "/book/abc123");
    expect(link).toHaveTextContent(product.name);
  });
});
