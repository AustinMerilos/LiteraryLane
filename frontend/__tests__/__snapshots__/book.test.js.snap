// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book/> Renders and matches the snapshot 1`] = `
<div>
  <div
    class="sc-hLQSwg jYDWXO"
  >
    <img
      alt="dogs are best"
      src="dog.jpg"
    />
    <h3
      class="sc-eDLKkx fBAZox"
    >
      <a
        href="/book/abc123"
      >
        dogs are best
      </a>
    </h3>
    <span
      class="sc-jTQCzO dwbbrB"
    >
      50.00
    </span>
    <div
      class="sc-gLLuof dOUzwr"
    />
  </div>
</div>
`;
